### ISSUE TITLE
Allow to define patterns against absolute file path

### ISSUE THREAD

**Issue by tamasmagedli** [depcheck/depcheck#107]
First of all it's a great tool and I wanted to integrate it into my project, though the use case of mine is a little bit different how you prefer to use it. Acutally I wanted to parse only a specific file list within the root directory so ignoring dirs doesn't help.

Reading the codebase I realised that I have another option, namely I can define parsers for patterns which is a cool a feature though it cannot be used for my use case, at least not for now.

Let's assume that we have two JS files within the `src` folder, and I want to use this pattern:
`src/*.js`

Currently it does not work as you get the basename from the absolute path of the file.
line 124 in check.js: 
`const basename = path.basename(filename);`
This prevents to use a prefix pattern for the absolute path like mentioned before.

Can you please omit the basename call and pass the filename to the minimatch?

Thank you for fixing this.

**Comment by lijunle** [comment]
I think this is a good change and all existing globs can be backward compatible because the `*` in `*.js` will match the dirname part.

Could you please send a pull request to implement it?

**Comment by lijunle** [comment]
Besides, it could be great to add one more test case to support the feature. :D

**Comment by tamasmagedli** [comment]
Yeah, I'll do that with new tests just wanted to know your opinion on this prior the changes.

**Comment by lijunle** [comment]
Closed by #108.
